### PR TITLE
Thread Safe implementation of PremailerContact

### LINKED ISSUE #6: This wrapper isn't thread safe
There is a race condition in PremailerContact class in JRuby. This class has a state in it i.e., the Permailer instance housed in `@pr`

When init it called this variable is initialized with the given string and options. Meanwhile the other two methods inline_css and plain_text use this variable to return the appropriate string. Since the state is initialized in a separate call `init` this results in a race condition when two threads execute the code such that:

Thread 1 -> init
Thread 2 -> init
Thread 1 -> inline_css
Thread 2 -> inline_css

This will result in re-initialization of the Premailer Instance with the data sent by Thread 2 in init. Which results in having incorrect values returned from inline_css or plain_text or other exceptions related to JRuby initialization (dependency loading etc.)

The fix would be remove the state from this class.

I have an PR in process - will be sending it in some time.

### PR DESCRIPTION
**Pull request by faisalferoz**
- Fixes #6
- Removes the init method from PremailerInterface

- Removes state from PremailerCcontact class

- Synchronizes methods of PremailerContact class as creating new instance of Premailer does dynamic loading of
classes (jars) which isn't thread safe

- Adds unit tests

### CODE REVIEW

**Comment by mdedetrich** [comment]
@faisalferoz Thanks for the PR, sorry for taking a while to look into it. I am on holdays but I will look into it in the next couple of weeks.

**Comment by faisalferoz** [comment]
@mdedetrich any update on this?

File: .gitignore [hunk 1]
```diff
@@ -25,4 +25,8 @@ buildNumber.properties
 # virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
 hs_err_pid*
 
+# eclipse
```
Comment (mdedetrich) [inline .gitignore]: Can you please put IDE specific settings in your own global `.gitignore`, see https://help.github.com/en/articles/ignoring-files#create-a-global-gitignore

**Review by mdedetrich — CHANGES_REQUESTED** [review]
Just remove the IDE specific settings from the .gitignore. Also do you know why the maven tests are failing?

File: .gitignore (hunk 1 above)
Comment (faisalferoz) [reply .gitignore]: Removed.

Comment (faisalferoz) [reply .gitignore]: I am not sure if these tests were passing before.

Comment (mdedetrich) [reply .gitignore]: Okay fair enough I will fix them later, I will merge and publish tomorrow then

**Review by mdedetrich — APPROVED** [review]

**Comment by mdedetrich** [comment]
@faisalferoz Released as `1.2_1.8.7`

**Pull request merged**